src/gui/rss/automatedrssdownloader.cpp

```diff
@@ -597,6 +587,7 @@ void AutomatedRssDownloader::handleFeedCheckStateChange(QListWidgetItem *feed_it
     updateMatchingArticles();
 }
 
+// FIXME: This should return early if any of the filters are invalid
```

Comment:
If multiple rules are selected then the valid filters can still match. The matching itself shortcuts if the filters aren't valid.

Also might want to look at PRs #6166 and #6169.